### PR TITLE
[FEA] Geometry dataloader

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
This PR adds a DALI based dataloader to work with Modulus Sym geometries. This can be used to sample point clouds on the surface or interior of geometry objects generated using Modulus' CSG module or ingested via STLs. 

Below shows an example of point cloud on the surface of ahmed body dataset. 

image

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus-sym/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus-sym/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus-sym/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/sym/geometry/geometry_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                shuffle=self.shuffle,
+                process_rank=self.process_rank,
+                world_size=self.world_size,
+            )
+
+            self.length = len(source) // self.batch_size
+            vars_tuple = dali.fn.external_source(
+                source,
+                num_outputs=len(self.requested_vars),
+                parallel=False,  # Parallel=False to simplify the code, you can switch to True if your source is thread-safe
```
Comment (mnabian) [inline modulus/sym/geometry/geometry_dataloader.py:168]: I suggest exposing this through the datapipe API

**Review by mnabian — APPROVED** [review]
LGTM. Left one comment

File: modulus/sym/geometry/geometry_dataloader.py (hunk 1 above)
Comment (ktangsali) [reply modulus/sym/geometry/geometry_dataloader.py]: Good catch. fixed

**Pull request merged**